query_optimizer/logical/HashJoin.hpp

```diff
@@ -50,9 +54,29 @@ typedef std::shared_ptr<const HashJoin> HashJoinPtr;
  */
 class HashJoin : public BinaryJoin {
  public:
+  enum JoinType {
+    kInnerJoin = 0,
+    kLeftSemiJoin,
+    kLeftAntiJoin
+  };
+
+
   LogicalType getLogicalType() const override { return LogicalType::kHashJoin; }
 
-  std::string getName() const override { return "HashJoin"; }
+  std::string getName() const override {
+    switch (join_type_) {
+      case kInnerJoin:
+        return "HashJoin";
+      case kLeftSemiJoin:
+        return "HashLeftSemiJoin";
+      case kLeftAntiJoin:
+        return "HashLeftAntiJoin";
+      default:
+        LOG(FATAL) << "Invalid JoinType: "
+                   << static_cast<typename std::underlying_type<JoinType>::type>(join_type_);
+    }
+    QUICKSTEP_UNREACHABLE();
```

Comment:
I think we could delete this line, as we either return a value in all valid conditions, or abort the program by default.